internal/datanode/flush_manager.go

```diff
@@ -175,9 +174,31 @@ func (q *orderFlushQueue) enqueueDelFlush(task flushDeleteTask, deltaLogs *DelDa
 // send into injectCh in there is running task
 // or perform injection logic here if there is no injection
 func (q *orderFlushQueue) inject(inject *taskInjection) {
-	q.injectCh <- inject
+	q.injectMut.Lock()
+	defer q.injectMut.Unlock()
+	// check if there are running task(s)
+	// if true, just put injection into injectCh
+	// in case of task misses an injection, the injectCh shall be drained in `postTask`
+	if q.runningTasks > 0 {
+		q.injectCh <- inject
+		return
+	}
+	// otherwise just handle injection here
+
+	q.handleInject(inject)
 }
 
+func (q *orderFlushQueue) handleInject(inject *taskInjection) {
+	// notify one injection done
+	inject.injectOne()
+	ok := <-inject.injectOver
```

Comment:
injectOver can be replaced by a barrier, if we just want to ensure all flush task reached end